### PR TITLE
Vi monodir

### PR DESCRIPTION
**Pull request by Vilax**
MonoDir (method for computing the directional resolution) included as protocol

A bug in add noise was corrected

B-factor correction now weights the filter by FSC values

### CODE REVIEW

**Comment by dmaluenda** [comment]
This PR is totally related to [this in Xmipp software](https://github.com/I2PC/xmipp/pull/22)

Please, **review, approve and merge** considering both!

File: xmipp3/protocols/protocol_preprocess/protocol_add_noise.py [hunk 1]
```diff
@@ -278,6 +280,9 @@ def getFileNameNoisyStk(self):
 
     def _updateParticle(self, particle, row):
         #fn = particle.getFileName()
-        fnOut = self.getFileNameNoisyStk()
-        particle.setFileName(fnOut)
+        # fnOut = self.getFileNameNoisyStk()
+        # particle.setFileName(fnOut)
+        from pyworkflow.em.packages.xmipp3.convert import xmippToLocation
```
Comment (dmaluenda) [inline xmipp3/protocols/protocol_preprocess/protocol_add_noise.py]: this is not fine. It Should be `from xmipp3.convert import xmippToLocation` and, please, put it at top of the code with the other imports

File: xmipp3/viewers/viewer_resolution3d.py [hunk 2]
```diff
@@ -187,7 +187,12 @@ def _applyBfactor(self, e=None):
         pixelSize = vol.getSamplingRate()
         args += '--sampling %f ' % pixelSize
         args += '--maxres %f ' % maxres
+        from os.path import exists
```
Comment (dmaluenda) [inline xmipp3/viewers/viewer_resolution3d.py]: imports should be at top of the code with the other imports

File: xmipp3/protocols/protocol_preprocess/protocol_add_noise.py (hunk 1 above)
Comment (Vilax) [reply xmipp3/protocols/protocol_preprocess/protocol_add_noise.py]: The line has been fixed, it was put at the top

File: xmipp3/viewers/viewer_resolution3d.py (hunk 2 above)
Comment (Vilax) [reply xmipp3/viewers/viewer_resolution3d.py]: the import has been fixed, now it appears at the top

**Comment by Vilax** [comment]
All your requests were solved

Tests exist, for MonoDir, and add noise.

**Pull request merged**